packages/core/saved-objects/core-saved-objects-migration-server-internal/src/document_migrator/document_migrator.ts

```diff
@@ -181,9 +184,10 @@ export class DocumentMigrator implements VersionedTransformer {
       const currentVersion = doc.typeMigrationVersion ?? doc.migrationVersion?.[doc.type];
       const latestVersion = this.migrations[doc.type].latestVersion[TransformType.Migrate];
       if (!allowDowngrade) {
-        throw Boom.badData(
+        const newerVersionModelError = createNewerModelVersionError(
           `Document "${doc.id}" belongs to a more recent version of Kibana [${currentVersion}] when the last known version is [${latestVersion}].`
         );
+        throw Boom.boomify(newerVersionModelError, { message: '[NewerModelVersionError]' });
```

Comment:
NIT: using an helper to create the error is fine, but I would move the whole logic into it

```ts
export function createNewerModelVersionError(message: string) {
  return Boom.boomify(Boom.badData(message), { message: '[NewerModelVersionError]' });
}

// ...

if (!allowDowngrade) {
   throw createNewerModelVersionError(`Document "${doc.id}" belongs to a more recent version of Kibana [${currentVersion}] when the last known version is [${latestVersion}].`)
}
```